test/mocks/thread_local/mocks.h

```diff
@@ -46,6 +46,7 @@ class MockInstance : public Instance {
   struct SlotImpl : public Slot {
     SlotImpl(MockInstance& parent, uint32_t index) : parent_(parent), index_(index) {
       parent_.data_.resize(index_ + 1);
+      parent_.deferred_data_.resize(index_ + 1);
```

Comment:
Instead of making this general here for one test, I would just override the behavior in the test by configuring an EXPECT_CALL, storing the value, etc. Then you can leave the default/simple behavior here.

Reply:
I originally wanted to do that, but I'm not sure if that's possible as SlotImpl is not a mock class..
i.e. when `SlotImpl::set()` is called in the FileConfigImpl during the , i don't think i have any way of changing the behaviour with EXPECT_CALL

Reply:
Hi Matt,  other components also need such MockThreadLocal with deferred call to test their code.   I think adding the code here will benefit these components.  Otherwise each component need to their own EXPECTED_CALL.

Reply:
We need to fix this in some better way. I'm really not happy with this type of one off fix for this issue which is going to keep happening in most other places. I'm very busy this week but will try to think of something in the next few days.